compiler/rustc_hir_typeck/src/gather_locals.rs

```diff
@@ -144,7 +144,9 @@ impl<'a, 'tcx> Visitor<'tcx> for GatherLocalsVisitor<'a, 'tcx> {
             let var_ty = self.assign(p.span, p.hir_id, None);
 
             if let Some((ty_span, hir_id)) = self.outermost_fn_param_pat {
-                if !self.fcx.tcx.features().unsized_fn_params {
+                if !(self.fcx.tcx.features().unsized_fn_params
+                    || self.fcx.tcx.features().repr_scalable)
```

Comment:
Does this not cause us to accept *regular* unsized fn params as well once `repr_scalable` is enabled? That isn't correct.